### PR TITLE
Python ToolCall新增函数、注解解析

### PR DESCRIPTION
**Pull request by jundaix**
更新tool call简化，添加了BaseModel，参数类型/注释缺失验证、参数描述提取，优先级等功能，补充了相关单测

### CODE REVIEW

File: python/tests/test_tool_definition_decorator.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+@unittest.skipUnless(os.getenv("TEST_CASE", "UNKNOWN") == "CPU_PARALLEL", "")
+class TestToolDefinitionDecorator(unittest.TestCase):
+    def test_disable_docstring(self):
+        @function(description="anotated function", disable_docstring=True)
+        @function_parameter(name="param", description="a parameter", type="str", default_value="default_val")
+        @function_return(description="a result", default_value="default_result")
+        def func(param: str) -> str:
+            return param
+
+        view = func.__pf_function__
```
Comment (wolvever) [inline python/tests/test_tool_definition_decorator.py:26]: __pf_function__ should change to ab_tool or sth,  cause pf represents Kaiwu's framework abbr.

File: python/utils/tool_definition_decorator.py [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+
+        _name_to_params = {}
+        for item in parameters:
+            _name_to_params[item.name] = item
+        self._name_to_params = _name_to_params
+
+        super().__init__()
+
+
+def function(
```
Comment (wolvever) [inline python/utils/tool_definition_decorator.py:270]: Use `manifest` might be better, like:

```
@manifest(description="...")
func get_weather(location: str):
  ....

```

File: python/utils/func_utils.py [hunk 3]
```diff
@@ -48,4 +60,135 @@ def __call__(cls, *args, **kwargs):
...
+class ParametersModel(BaseModel):
+    type: Literal["object"]
+    properties: Dict[str, PropertyModel]
+    required: List[str]
+
+class FunctionModel(BaseModel):
+    type: Literal["function"]
+    function: Dict[str, Any]
+
+def function_to_model(func) -> FunctionModel:
```
Comment (wolvever) [inline python/utils/func_utils.py:78]: function_to_manifest

File: python/utils/func_utils.py [hunk 4]
```diff
@@ -48,4 +52,157 @@ def __call__(cls, *args, **kwargs):
...
+        type (Literal["object"]): 表示参数集合的类型，固定为 "object"。
+        properties (Dict[str, PropertyModel]): 参数的具体属性映射，其中键是参数名，值是对应的属性模型。
+        required (List[str]): 必须提供的参数列表。
+    """
+    type: Literal["object"]
+    properties: Dict[str, PropertyModel]
+    required: List[str]
+
+
+class FunctionModel(BaseModel):
```
Comment (wolvever) [inline python/utils/func_utils.py:79]: The naming of FunctionModel, func_utils, tool_definition_xxx, FunctionView and function_to_manifest is not consistent.
The core concepts we have here in AppBuilder are Component and tools parameter.  Each tool is a Component, or a combination of Components.

utils -> manifest
   func_utils -> models.py, which contains manifest object models that are Manifest and Parameter.  Manifest class has static methods like `from_function(func) -> Manifest`, and `from_openapi(spec_filepath | spec) -> Manifest`.
   
   tool_definition_decorator.py -> manifest_decorator.py
   tool_definition_signature.py ->  manifest_function_signature.py

File: python/utils/func_utils.py [hunk 5]
```diff
@@ -48,4 +52,157 @@ def __call__(cls, *args, **kwargs):
...
+
+class ParametersModel(BaseModel):
+    """函数参数模型，用于定义函数的参数结构。
+
+    Attributes:
+        type (Literal["object"]): 表示参数集合的类型，固定为 "object"。
+        properties (Dict[str, PropertyModel]): 参数的具体属性映射，其中键是参数名，值是对应的属性模型。
+        required (List[str]): 必须提供的参数列表。
+    """
+    type: Literal["object"]
```
Comment (wolvever) [inline python/utils/func_utils.py:74]: Use explicit name, description, type, required.  Leave properties for extension.

File: python/utils/tool_definition_signature.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    Any: "Any",
+    int: "int",
+    str: "str",
+    float: "float",
+    bool: "bool",
+    # 根据需要添加更多映射
+}
+
+
+def get_signature_view(func):
```
Comment (wolvever) [inline python/utils/tool_definition_signature.py:28]: get_signature

File: python/utils/func_utils.py [hunk 7]
```diff
@@ -48,4 +52,157 @@ def __call__(cls, *args, **kwargs):
...
+            "name": func.__name__,
+            "description": func.__doc__,
+            "parameters": parameters_model.model_dump(),
+            "returns": return_info,
+        },
+    )
+
+    return function_manifest
+
+def decorator_to_manifest(function_view) -> FunctionModel:
```
Comment (wolvever) [inline python/utils/func_utils.py:153]: Didn't find FunctionView class anywhere,  rename function_view

File: python/utils/func_utils.py [hunk 8]
```diff
@@ -48,4 +52,157 @@ def __call__(cls, *args, **kwargs):
...
+    """
+    将函数视图转换为函数模型。
+    
+    Args:
+        function_view (FunctionView): 函数视图对象。
```
Comment (wolvever) [inline python/utils/func_utils.py:158]: Also the docstring

File: python/core/manifest/models.py [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+    Attributes:
+        type (Literal["function"]): 表示模型的类型，固定为 "function"。
+        function (Dict[str, Any]): 函数的详细信息，包括名称、描述、参数、返回值等。
+    """
+    type: Literal["function"]
+    function: Dict[str, Any]
+
+def function_to_manifest(func) -> Manifest:
+    """
+    利用 tool_definition_signature.py 提供的 get_signature_view 方法解析函数的签名和参数信息，
```
Comment (userpj) [inline python/core/manifest/models.py:54]: `tool_definition_signature.py`、`get_signature_view`、`FunctionModel` not found in code

File: python/core/manifest/models.py [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+        example (str): The example of the parameter.
+        kind (str): The kind of the parameter. For python function it is argument or return.
+          For OpenAPI it is location like query, path, header, cookie, body.
+    """
+
+    name: str
+    description: str = None
+    default_value: str = None
+    type_: str = None
+    type_object: Optional[Any] = Field(default=None, metadata=config(exclude=_exclude))
```
Comment (userpj) [inline python/core/manifest/models.py]: metadata = config(exclude=lambda value: True)

File: python/core/manifest/manifest_decorator.py [hunk 11]
```diff
@@ -0,0 +1,350 @@
...
+
+    if hasattr(func, "__ab_manifest__"):
+        view = func.__ab_manifest__
+        return view.description if view else ""
+
+
+def update_function_description(func, description):
+    """Update function description."""
+    func.__dict__["__kernel_function_description__"] = description or ""
+    func.__dict__["__pf_func_description__"] = description or ""
```
Comment (userpj) [inline python/core/manifest/manifest_decorator.py:36]: __pf_func_description__

File: python/core/manifest/manifest_decorator.py [hunk 12]
```diff
@@ -0,0 +1,350 @@
...
+            default_value=default_value,
+            type_=type,
+            required=required,
+            example=example,
+            kind=ParameterViewKind.ARGUMENT,
+        )
+
+        # Update parameter view lists for function_parameter decorator.
+        # This will be merged into ManifestView if function decorator runs last like:
+        # @function
```
Comment (userpj) [inline python/core/manifest/manifest_decorator.py:233]: # @manifest
# @manifest_parameter
# @manifest_parameter

File: docs/BasisModule/Platform/Application/appbuilder_client.md [hunk 13]
```diff
@@ -279,6 +285,130 @@ msg_2 = client.run(
...
+  return "北京今天25度"
+  
+#定义函数列表
+functions = [get_current_weather]
+function_map = {f.__name__: f for f in functions}
+#调用大模型
+msg = client.run(
+  conversation_id=conversation_id,
+  query="今天北京的天气怎么样？",
+  tools = [appbuilder.function_to_manifest(f).model_dump() for f in functions]
```
Comment (wolvever) [inline docs/BasisModule/Platform/Application/appbuilder_client.md:326]: tools = [Manifest.from(f) for f in functions]

File: python/core/manifest/manifest.py [hunk 14]
```diff
@@ -0,0 +1,124 @@
+from appbuilder.core.manifest.models import Manifest, ParametersModel, PropertyModel
+from appbuilder.core.manifest.manifest_signature import get_signature
+from appbuilder.core.manifest.manifest_decorator import ManifestView
+
+def function_to_manifest(func) -> Manifest:
```
Comment (wolvever) [inline python/core/manifest/manifest.py:5]: Move to Manifest class

**Pull request closed without merging**